src/torchrunx/agent.py

```diff
@@ -93,13 +93,14 @@ def entrypoint(serialized_worker_args: bytes):
 
     logger.debug(f"executing function: {worker_args.function}")
 
-    r = worker_args.function()
-
-    # flush streams
-    sys.stdout.flush()
-    sys.stderr.flush()
-
-    return r
+    try:
+        return worker_args.function()
+    except Exception as e:
+        logger.error(e)
+        return WorkerException(exception=e)
+    finally:
```

Comment:
except: … raise
finally: return return_object

Reply:
Is there any point to raising the exception?

Reply:
I guess just that actually raising the exception (which should be logged via redirect_stderr anyway) is more "proper" than logger.log(exception)? There may be a difference in the log output?